library/common/jni/BUILD

```diff
@@ -36,8 +37,64 @@ cc_binary(
     ],
     linkshared = True,
     deps = [
+        ":android_log",
         ":jni_utility_lib",
         "//library/common:envoy_main_interface_lib",
         "//library/common/api:c_types",
     ],
 )
+
+# Java JNI library
+cc_library(
+    name = "envoy_java_jni_lib",
+    srcs = [
+        "jni_interface.cc",
+        "jni_utility.cc",
+        "jni_version.cc",
+        "@local_jdk//:jni_header",
+    ],
+    hdrs = [
+        "jni_utility.h",
+        "jni_version.h",
+    ],
+    copts = ["-std=c++14"],
+    linkopts = [
+        "-lm",
+    ],
+    deps = [
+        "java_log",
+        "//library/common:envoy_main_interface_lib",
+        "//library/common/types:c_types_lib",
+        "@bazel_tools//tools/jdk:jni",
+        "@envoy//source/common/common:assert_lib",
+    ],
+)
+
+# Utility implementation for Java JNI
+cc_library(
+    name = "java_log",
+    srcs = [
+        "java_jni_support.cc",
+        "@local_jdk//:jni_header",
+    ],
+    hdrs = ["jni_support.h"],
+    copts = ["-std=c++14"],
+    linkopts = [
+        "-lm",
+    ],
+    deps = ["@bazel_tools//tools/jdk:jni"],
+)
+
+# Utility implementation for NDK JNI
+cc_library(
+    name = "android_log",
```

Comment:
nit: can we update this bazel target to match the name of the source files, i.e. "ndk_jni_support"?